sandbox/rocky/tf/plotter/plotter.py

```diff
@@ -0,0 +1,108 @@
+import atexit
+from collections import namedtuple
+import enum
+from enum import Enum
+from multiprocessing import Process
+import platform
+from queue import Empty, Queue
+from threading import Thread
+
+import numpy as np
+import tensorflow as tf
+
+from rllab.sampler.utils import rollout
+
+__all__ = ["Plotter"]
+
+
+class Op(Enum):
+    STOP = 0
+    UPDATE = 1
+    DEMO = 2
+
+
+Message = namedtuple("Message", ["op", "args", "info"])
```

Comment:
info? the python convention would be kwargs (ignore what gym does)

Reply:
To add more color, the intention of the interface is this.

All python function definitions can be written as
```python
def foo(*args, **kwargs):
    # do some stuff
```
where `args` is an ordered list of unnamed arguments and `kwargs` is a dictionary of keyword arguments. You can even call any function like this, even if it wasn't declared this way.
```
def foo(bar, baz=0):
    return bar + baz

args = [1]
kwargs = {"baz": 1}
print(foo(*args, **kwargs)) # prints 2
```

Now imagine that this interface gets complicated, and we split the implementation of DEMO, UPDATE, and STOP into functions.
If `Message` has op, args, and kwargs, we can implement dispatch for *any function*:
```python
if Op.FOO in msgs:
    msg = msgs[Op.FOO]
    do_foo(*msg.args, **msg.kwargs)
```